File: Backend/MedicineDelivery/MedicineDelivery.Domain/Entities/Consent.cs
namespace MedicineDelivery.Domain.Entities
{
    public class Consent
    {
        public Guid ConsentId { get; set; } = Guid.NewGuid();
        public string Title { get; set; } = string.Empty;
        public string? Description { get; set; }
        public string Content { get; set; } = string.Empty;
        public bool IsActive { get; set; } = true;
        public DateTime CreatedOn { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedOn { get; set; }

        // Navigation property
        public ICollection<ConsentLog> ConsentLogs { get; set; } = new List<ConsentLog>();
    }
}

File: WebApp/src/models/Consent.ts
export interface Consent {
  consentId: string;
  title: string;
  description?: string;
  content?: string;
  isActive: boolean;
  createdOn?: string;
}

export interface ConsentLog {
  consentLogId: string;
  consentId: string;
  userId: string;
  userType?: string;
  respectiveId?: string;
  action: number; // 1 = Accept, 2 = Reject
  userAgent?: string;
  ipAddress?: string;
  deviceInfo?: string;
  createdOn?: string;
}
